agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/processors/AttributeProcessor.java

```diff
@@ -167,6 +169,26 @@ private static SpanData processExtractAction(SpanData span, ProcessorAction acti
     return new MySpanData(span, builder.build());
   }
 
+  private static SpanData processMaskAction(SpanData span, ProcessorAction actionObj) {
+    // Currently we only support String
+    String existingValue = span.getAttributes().get(actionObj.key);
+    if (existingValue == null) {
+      return span;
+    }
+    Matcher matcher = actionObj.maskAttribute.pattern.matcher(existingValue);
+    if (!matcher.matches()) {
+      return span;
+    }
+    AttributesBuilder builder = span.getAttributes().toBuilder();
+    String newValue = actionObj.maskAttribute.replace;
+    for (String groupName : actionObj.maskAttribute.groupNames) {
+      newValue = newValue.replaceAll(groupName, matcher.group(groupName));
+    }
+    newValue = newValue.replaceAll("\\$", "");
```

Comment:
wondering if this works, relying on Java regular expression's own replacement mechanism (may need to change replacement to `${name}`, e.g. `"${uriNoCard}****${cardEnd}"`)

```suggestion
    Matcher matcher = actionObj.maskAttribute.pattern.matcher(existingValue);
    String newValue = matcher.replaceAll(actionObj.maskAttribute.replace);
    if (newValue.equals(existingValue)) {
      return span;
    }
    AttributesBuilder builder = span.getAttributes().toBuilder();
```

Reply:
thanks for the suggestion. It looks more clean now